src/Features/Core/Portable/ExternalAccess/VSTypeScript/Api/VSTypeScriptCompletionProvider.cs

```diff
@@ -15,12 +15,13 @@ internal abstract class VSTypeScriptCompletionProvider : CompletionProvider
         public sealed override bool ShouldTriggerCompletion(SourceText text, int caretPosition, CompletionTrigger trigger, OptionSet options)
         {
             Debug.Fail("For backwards API compat only, should not be called");
-            return ShouldTriggerCompletionImpl(text, caretPosition, trigger, CompletionOptions.Default.TriggerOnTypingLetters);
+            var defaultOptions = CompletionOptions.Default;
+            return ShouldTriggerCompletionImpl(text, caretPosition, trigger, defaultOptions.TriggerOnTypingLetters, defaultOptions.ExpandedCompletionBehavior);
         }
 
         internal sealed override bool ShouldTriggerCompletion(HostLanguageServices languageServices, SourceText text, int caretPosition, CompletionTrigger trigger, CompletionOptions options, OptionSet passThroughOptions)
-            => ShouldTriggerCompletionImpl(text, caretPosition, trigger, options.TriggerOnTypingLetters);
+            => ShouldTriggerCompletionImpl(text, caretPosition, trigger, options.TriggerOnTypingLetters, options.ExpandedCompletionBehavior);
 
-        protected abstract bool ShouldTriggerCompletionImpl(SourceText text, int caretPosition, CompletionTrigger trigger, bool triggerOnTypingLetters);
+        protected abstract bool ShouldTriggerCompletionImpl(SourceText text, int caretPosition, CompletionTrigger trigger, bool triggerOnTypingLetters, ExpandedCompletionMode completionMode);
     }
```

Comment:
Instead of changing signature of the existing method, we can add a new virtual method:

```
protected virtual bool ShouldTriggerCompletionImpl(SourceText text, int caretPosition, CompletionTrigger trigger, bool triggerOnTypingLetters, VSTypeScriptExpandedCompletionMode completionMode)
 => ShouldTriggerCompletionImpl(text, caretPosition, trigger, triggerOnTypingLetters);
```

We can make the change in Roslyn first and then TypeScript can override the new method in an independent insertion.

Reply:
We don't need another method nor change the signature, I think [this approach might work](https://github.com/dotnet/roslyn/pull/60635/files#r845612810).